### PR TITLE
feat: add efficientnetv2 models

### PR DESCRIPTION
**Pull request by yangobeil**
This PR changes from using an external package to using the EfficientNet models included in tensorflow. The new EfficientNetV2 models are also now made available to use.

An important change to note is that images are not resized to (299,299) anymore when they are transformed into tfrecords. This is because we realized that different models have different recommended input size so they are taken into account when training.

Various small fixes have been made at different places to account for the changes. An important one is that the tensorflow EfficientNets devide the pixels by 255 directly in the model so the operation should not be done in the data preprocessing.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -25,13 +25,13 @@ A great way of explaining the library is using an example notebook, which you ca
 
 The most recent version of this library adds a feature to leverage unlabelled images in order to improve the performance
 of image classifiers. This procedure is called semi-supervised learning (SSL) and is discussed in this `blog post <https://medium.com/decathlondevelopers/improving-performance-of-image-classification-models-using-pretraining-and-a-combination-of-e271c96808d2/>`_.
+The method was also described in a `paper <https://arxiv.org/abs/2108.08362/>`_ and presented at a conference.
```
Comment (hasibzunair) [inline docs/index.rst]: Maybe we can add a [link](https://drive.google.com/drive/folders/178j1L8wivD5h1q3wkfs0XbXPNpf2LdyS) to the presentation?

**Review by hasibzunair — DISMISSED** [review]
Went over them. Looks good!

File: docs/index.rst [hunk 2]
```diff
@@ -25,7 +25,7 @@ A great way of explaining the library is using an example notebook, which you ca
 
 The most recent version of this library adds a feature to leverage unlabelled images in order to improve the performance
 of image classifiers. This procedure is called semi-supervised learning (SSL) and is discussed in this `blog post <https://medium.com/decathlondevelopers/improving-performance-of-image-classification-models-using-pretraining-and-a-combination-of-e271c96808d2/>`_.
-The method was also described in a `paper <https://arxiv.org/abs/2108.08362/>`_ and presented at a conference.
+The method was also described in a `paper <https://arxiv.org/abs/2108.08362/>`_ and presented at a `conference <http://mmsports.multimedia-computing.de/mmsports2021/program.html>`_.
```
Comment (hasibzunair) [inline docs/index.rst]: Maybe we can add:

..presented at ACM MMSports 2021

Comment (yangobeil) [reply docs/index.rst]: Good idea. I included your suggestions.

**Review by hasibzunair — APPROVED** [review]

**Pull request merged**